### DISCUSSION TITLE
Use xonsh subprocess expressions in python

### DISCUSSION THREAD

**Discussion by FlyingWombat** [xonsh/xonsh#5406] [Q&A]
The docs in [Tutorial: xonsh projects](https://xon.sh/tutorial_xonsh_projects.html) say that xonsh can be used interoperably with Python. I tried to import xonsh from python, using the instructions on that page, and get a syntax error when I try to execute a subprocess expression.

I get a Syntax error.
```python
#!/usr/bin/env python3

from xonsh.main import setup
setup()
del setup

print($(ls))
```
```
$  ./xonsh-test.py
  File "/home/wombat/./xonsh-test.py", line 7
    print($(ls))
          ^
SyntaxError: invalid syntax
```

Can someone explain what I'm missing? Or am I misunderstanding that doc page, and this isn't one of xonsh intended use cases?

**Comment by anki-code** [comment]
There are two ways:
* Run xonsh script via xonsh i.e. `xonsh script.xsh` (your case). You can use any xonsh syntax in the `xsh` file.
* Use pure python xonsh library i.e. `import xonsh` in `script.py` and run `python script.py`. In this case you're using xonsh package with functions without xonsh syntax.

Take a look into [xonsh cheatsheet](https://github.com/anki-code/xonsh-cheatsheet/).
